packages/flutter/test/rendering/editable_test.dart

```diff
@@ -367,4 +367,59 @@ void main() {
     expect(currentSelection.baseOffset, 1);
     expect(currentSelection.extentOffset, 3);
   });
+
+  test('selection does not flicker as user is dragging', () {
```

Comment:
Add this test to material and cupertino text_field_test too via gestures (in case we move those widgets to use something else than selectPositionAt in the future).

Reply:
Added the test to material for now. Cupertino still doesn't have text selection via mouse but I'll have a PR for that tomorrow.

Reply:
Is this a mouse specific issue or would this not happen with a handle drag?

Reply:
@xster yes, for some reason dragging the handles doesn't flicker at all.

That said, your comment made me think about other scenarios that may cause flickering, and I found one :) Long-press-drag has the same issue in the material text field. But it's not fixed by this PR. Filed an issue here to be fixed separately: https://github.com/flutter/flutter/issues/29682.